ThreeBotPackages/threefold/tfgrid_workloads/actors/workload_manager.py

```diff
@@ -174,13 +199,32 @@ def reservation_register(self, reservation, schema_out=None, user_session=None):
         reservation = (O) !tfgrid.reservation.1
         ```
         """
+        workloads = [l for _, l in self._iterate_over_workloads(reservation)]
```

Comment:
Also need to check that there is not duplicate in the IDs.
I would  also create a function `reservation_validate` and move all the logic regarding validation in it. It will makes things easier to read and to maintains in the future if we have all the logic in one place.